spec.bs

```diff
@@ -1101,13 +1303,81 @@ integer |maxAllowed|, perform the following steps. They return an integer.
     the result of adding |returnValue| to |value|["{{PASignalValue/offset}}"].
 1. Clamp |returnValue| to the range [0, |maxAllowed|] and return the result.
 
-Issue: Maybe add refs to the rounding logic.
-
-To <dfn>determine the numeric value</dfn> of a [=signal base value=]
-<var ignore=''>signalBaseValue</var>, perform the following steps. They return a
-{{double}}.
-
-Issue: Fill in.
+To <dfn>determine a signal's numeric value</dfn> given a [=signal base value=]
+|signalBaseValue| and a <a spec="turtledove">leading bid info</a>
+|leadingBidInfo|, perform the following steps. They return a {{double}}.
+1. If |signalBaseValue| is "<code>[=signal base value/winning-bid=]</code>":
+    1. If |leadingBidInfo|'s <a spec="turtledove" for="leading bid info">leading
+        bid</a> is null, return 0.
+    1. Otherwise, return  |leadingBidInfo|'s
+        <a spec="turtledove" for="leading bid info">leading bid</a>'s
+        <a spec="turtledove" for="generated bid">bid</a>.
+1. If |signalBaseValue| is
+    "<code>[=signal base value/highest-scoring-other-bid=]</code>":
+    1. If |leadingBidInfo|'s <a spec="turtledove" for="leading bid info">highest
+        scoring other bid</a> is null, return 0.
+    1. Otherwise, return |leadingBidInfo|'s
+        <a spec="turtledove" for="leading bid info">highest scoring other
+        bid</a>'s <a spec="turtledove" for="generated bid">bid</a>.
+1. If |signalBaseValue| is "<code>[=signal base value/script-run-time=]</code>":
+    1. Return the number of milliseconds of CPU time that the calling function
+        (e.g. `generateBid()`) took to run.
+1. If |signalBaseValue| is
+    "<code>[=signal base value/signals-fetch-time=]</code>":
+    Switch on the associated [=worklet function=]:
+    <dl class="switch">
+    : <code>[=worklet function/generate-bid=]</code>
+    :: Return the number of milliseconds it took for the trusted bidding signals
+        fetch to complete, or 0 if no fetch was made.
+    : <code>[=worklet function/score-ad=]</code>
+    :: Return the number of milliseconds it took for the trusted scoring signals
+        fetch to complete or 0 if no fetch was made.
+    : <code>[=worklet function/report-result=]</code>
+    : <code>[=worklet function/report-win=]</code>
+    :: Return 0.
+
+    </dl>
+
+    Issue: Consider disallowing this in the latter two [=worklet functions=].
+1. If |signalBaseValue| is
+    "<code>[=signal base value/bid-reject-reason=]</code>":
+    1. If the bid did not succeed purely because it didn't meet the required
+        k-anonymity threshold, return 8.
```

Comment:
maybe useful to export the ["k-anonymity threshold" definition](https://github.com/WICG/turtledove/blob/main/spec.bs#LL1401C43-L1401C43) from Protected Audience spec and add a autolink here using `[=k-anonymity threshold=]`? I can export that if you'd like to link to it.

Reply:
Added this link :)